consensus/istanbul/backend/announce.md

```diff
@@ -0,0 +1,140 @@
+
+## Nearly Elected Validator (NEV)
+
+A validator is a Nearly Elected Validator (NEV) if and only if:
+
+- It is a current elected validator in this epoch, or
+- It is a validator in the result set from calling `ElectNValidatorSigners` with `additionalAboveMaxElectable` taking the value of 10.
+
+## Spec by node type
+
+Note that validators are full nodes too, therefore they should follow the same rules as full nodes. This is specially important to not give away the status of the node as a validator.
+
+### Full node (non validator) Spec
+
+#### Peer handshake
+
+During an inbound peer connection, the remote peer can send an [enodeCertificateMsg] to identify itself as a validator. This allows for preferential treatment for the p2p connection.
+
+#### Peer registered
+
+When a peer is registered, all highest known version certificates should be sent to the registered peer in a [versionCertificatesMsg].
+
+#### Handling [queryEnodeMsg]
+
+Messages received of this type should be only processed once.
+Should be regossipped as is, unless another message from the same validator origin has been regossipped in the past 5 minutes.
+
+#### Handling [enodeCertificateMsg]
+
+[enodeCertificateMsg] should be ignored by non validators.
+
+#### Handling [versionCertificatesMsg]
+
+Messages received of this type should be only processed once.
+When received, regossips a new [versionCertificatesMsg] with certificates that haven't been gossiped (identified by validator address only) in the past 5 minutes
+
+#### Message spawning
+
+Every 5 minutes it should regossip all the highest known version certificates for all the [NearlyElectedValidator]
+
+### Validator Spec
+
+#### Replying [queryEnodeMsg]
+
+Should be replied with an [enodeCertificateMsg] if the origin is a validator and this validator is the destination, and if an already existing connection to the origin validator exists.
+
+#### Receiving [enodeCertificateMsg]
+
+Should update the highest known version to the one given, even if there's no version certificate for it known by this validator.
+
+#### Query spawning
+
+One minute (60 seconds) after a validator enters the set of [NearlyElectedValidator], it should start sending [queryEnodeMsg] messages to the network, for all other NEV validators that have a higher version `eNodeURL` than the one currently known.
+
+Messages should be spaced by at least 5 minutes (300 seconds). A query for a specific `<validator, version>` tuple has a retry back off period. The `nth` attempt should be spaced from the previous one by:
+
+```
+timeoutMinutes = 1.5 ^ (min(n - 1, 5))
+```
+
+Optionally, the first ten (10) query messages can be spaced by at least 60 seconds and ignoring retry back off periods for unanswered queries. This is known as the `AggressiveQueryEnodeGossip`.
+
+If the validator is no longer in the list of [NearlyElectedValidator] then it should stop sending [queryEnodeMsg] messages.
```

Comment:
I think we can drop this statement, non NEV nodes are not validators so at that point it behaves as a full node.

Reply:
This raises the question of 'what is a validator' for the purpose of this document then. I was using the definition of the istanbul backend code which is if the geth node was started with the `--mine` argument. So it can be a validator, but maybe it doesn't have enough votes to be a NEV (or much less an elected validator)

Reply:
Ahh right, that seems reasonable but, would be good to just have a section that clarifies the terminology.